eva_submission/eload_migration.py

```diff
@@ -0,0 +1,66 @@
+import os
+import shutil
+import subprocess
+
+import yaml
+from ebi_eva_common_pyutils import command_utils
+from ebi_eva_common_pyutils.config import cfg
+
+from eva_submission import NEXTFLOW_DIR
+from eva_submission.eload_submission import Eload
+
+
+class EloadMigration(Eload):
+
+    def migrate(self, project_accession=None):
+        self.run_nextflow_copy(project_accession)
+        self.update_and_reload_config()
+
+    def run_nextflow_copy(self, project_accession=None):
+        migrate_params = {
+            'eload': self.eload,
+            'old_eloads_dir': cfg['old_eloads_new_mnt'],
+            'new_eloads_dir': cfg['eloads_dir'],
+            'old_projects_dir': cfg['old_projects_new_mnt'],
+            'new_projects_dir': cfg['projects_dir'],
+        }
+        if project_accession:
+            migrate_params['project_accession'] = project_accession
+        work_dir = self.create_nextflow_temp_output_directory()
+        params_file = os.path.join(self.eload_dir, 'migrate_params.yaml')
+        # Use a specific log file so we don't overwrite when we sync
+        log_file = os.path.join(self.eload_dir, 'migrate_nextflow.log')
+
+        with open(params_file, 'w') as open_file:
+            yaml.safe_dump(migrate_params, open_file)
+        nextflow_script = os.path.join(NEXTFLOW_DIR, 'migrate.nf')
+
+        try:
+            command_utils.run_command_with_output(
+                f'Nextflow migrate process',
```

Comment:
Since rsync is essentially an idempotent/resumable operation anyway, I am wondering the rationale for the copy operations being Nextflow scripts. Is it to enforce them to be executed in the 'datamover' queue?

Reply:
Yes exactly, this is my attempt at being consistent with levels of coordination as we've discussed many times now... the trade-off didn't seem too bad in this case but I'm open to changing it.  (It's actually not even a resumable Nextflow process, since as you say rsync is idempotent/resumable anyway.)

It also lets Nextflow parallelise copying the eload and project directories, which is kind of nice.